### PR TITLE
converter: speed up conversion

### PR DESCRIPTION
**Pull request by imeoer**
```
Support the new option `--type tar-rafs` of builder for Pack method in converter
package, enables converting OCI tar blob stream into nydus blob directly,
eliminating the need to decompress it to a local directory first, thus
greatly accelerating the pack process.
```

Signed-off-by: Yan Song <imeoer@linux.alibaba.com>

### CODE REVIEW

**Comment by changweige** [comment]
Does this PR depend on nydusd and nydus-image 2.2+ ?

**Review by changweige — COMMENTED** [review]
Overall LGTM. Only a few tiny comments left

File: pkg/converter/tool/feature.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+var currentVersion string
+var currentVersionDetectOnce sync.Once
+var disableTar2Rafs = os.Getenv(envNydusDisableTar2Rafs) != ""
+
+var (
+	// The option `--type tar-rafs` enables converting OCI tar blob
+	// stream into nydus blob directly, the tar2rafs eliminates the
+	// need to decompress it to a local directory first, thus greatly
+	// accelerating the pack process.
+	FeatureTar2Rafs Feature = "--type tar-rafs"
```
[the remark this answers is not in the document]
Comment (imeoer) [reply pkg/converter/tool/feature.go:31]: Fixed

File: pkg/converter/convert_unix.go [hunk 2]
```diff
@@ -392,61 +405,86 @@ func packRef(ctx context.Context, dest io.Writer, opt PackOption) (io.WriteClose
 		}
 	}()
 
-	blobMetaPath := filepath.Join(workDir, "blob.meta")
-	blobMetaFifo, err := fifo.OpenFifo(ctx, blobMetaPath, syscall.O_CREAT|syscall.O_RDONLY|syscall.O_NONBLOCK, 0644)
+	rafsBlobPath := filepath.Join(workDir, "blob.rafs")
```
[the remark this answers is not in the document]
Comment (imeoer) [reply pkg/converter/convert_unix.go:408]: It's a temp file, just be used as a unique name.

File: pkg/converter/convert_unix.go [hunk 3]
```diff
@@ -392,61 +405,86 @@ func packRef(ctx context.Context, dest io.Writer, opt PackOption) (io.WriteClose
...
+	rafsBlobFifo, err := fifo.OpenFifo(ctx, rafsBlobPath, syscall.O_CREAT|syscall.O_RDONLY|syscall.O_NONBLOCK, 0644)
 	if err != nil {
 		return nil, errors.Wrapf(err, "create fifo file")
 	}
 
-	sourcePath := filepath.Join(workDir, "blob.targz")
-	sourceFifo, err := fifo.OpenFifo(ctx, sourcePath, syscall.O_CREAT|syscall.O_WRONLY|syscall.O_NONBLOCK, 0644)
+	tarBlobPath := filepath.Join(workDir, "blob.targz")
```
[the remark this answers is not in the document]
Comment (imeoer) [reply pkg/converter/convert_unix.go:414]: Ditto.

**Pull request merged**